src/scripts/components/imagepanel/select.js

```diff
@@ -0,0 +1,156 @@
+  constructor(props) {
+    super(props);
+    this.state = {
+      imageSize: props.imageSize,
+      imgHeight: props.imgHeight,
+      imgWidth: props.imgWidth,
+      resolutions: props.resolutions,
+      onSelectionChange: props.onSelectionChange,
+      fileType: props.fileType,
+      fileTypes: props.fileTypes,
+      fileSize: props.fileSize,
+      proj: this.props.proj,
+      worldfile: props.worldfile,
+      resolution: props.resolution,
+      valid: props.valid
+    };
+  }
+  handleChange(type, value) {
+    var fileType = this.state.filetype;
+    var resolution = this.state.resolution;
+    var worldfile = this.state.worldfile;
+    if (type === 'resolution') {
+      resolution = value;
+    } else if (type === 'worldfile') {
+      worldfile = value;
+    } else {
+      fileType = value;
+    }
+    this.state = {
+      resolution: resolution,
+      fileType: fileType,
+      worldfile: worldfile
+    };
+    this.props.onSelectionChange(resolution, worldfile, fileType);
+  }
+  _renderFileTypeSelect() {
+    if (this.props.fileTypeOptions) {
+      return (
+        <div className='wv-image-header'>
+          <SelectionList
+            id='wv-image-format'
+            optionName='filetype'
+            value={this.state.filetype}
+            optionArray={this.state.fileTypes}
+            onChange={this.handleChange.bind(this)}
+          />
+          {this.props.secondLabel}
+        </div>
+      );
+    }
+  }
+  _renderWorldfileSelect() {
+    if (this.props.worldFileOptions) {
+      return (
+        <div className='wv-image-header'>
+          <select
+            id='wv-image-worldfile'
+            onChange={(e) => this.handleChange('worldfile', e.target.value)}>
+            <option value='false'>No</option>
+            <option value='true'>Yes</option>
+          </select>
+          Worldfile (.zip)
+        </div>
+      );
+    }
+  }
+  render() {
+    var filetypeSelect = this._renderFileTypeSelect();
+    var worldfileSelect = this._renderWorldfileSelect();
+
+    return (
+      <div className="wv-re-pick-wrapper">
+        <div className='wv-image-header'>
+          <SelectionList
+            id='wv-image-resolution'
+            optionArray={this.state.resolutions}
+            value={this.state.resolution}
+            optionName='resolution'
+            onChange={this.handleChange.bind(this)}
+          />
+          {this.props.firstLabel}
+        </div>
+        {filetypeSelect}
+        {worldfileSelect}
+        <ResTable
+          width={this.state.imgWidth}
+          height={this.state.imgHeight}
+          fileSize={this.state.fileSize}
+          maxImageSize={this.props.maxImageSize}
+          valid={this.state.valid}
+          onClick={this.props.onDownloadClick}
+        />
+      </div>
+
+    );
+  }
+}
+
+ImageResSelection.defaultProps = {
+  firstLabel: 'Resolution (per pixel)',
+  secondLabel: 'Format',
+  width: '0000',
```

Comment:
Curious about the value of this prop being `0000`, likewise with `height` and `imageSize` values.